stix_shifter_modules/nozomi/stix_transmission/connector.py

```diff
@@ -0,0 +1,216 @@
+                            break
+                        # parsing the next_url
+                        page = regex.findall(r'page=([^>]*?)&', query_url)[0]
+                        next_url = regex.sub(r'page=([^>]*?)&', f'page={str(int(page) + 1)}&', query_url)
+
+                        if not metadata and result_count < total_records:
+                            remaining_records = total_records - result_count
+                        elif metadata and local_result_count < total_records:
+                            remaining_records = total_records - local_result_count
+                        else:
+                            break
+
+                        if remaining_records > Connector.NOZOMI_MAX_PAGE_SIZE:
+                            page_size = Connector.NOZOMI_MAX_PAGE_SIZE
+                        else:
+                            page_size = remaining_records
+                    # Handling expired token
+                    elif response_wrapper.code == 401 and not token_generated and \
+                            (response == '' or
+                             'Signature has expired' in response_dict.get('error', {}).get('message', '')):
+                        jwt_token = await self.__get_token()
+                        token_generated = True
+                        continue
+                    else:
+                        error = response_dict.get('error', '')
+                        if response_wrapper.code == 403:
+                            error = 'Query length is too long or Invalid Query'
+                        return_obj = self.exception_response(response_wrapper.code, error)
+                        data = []
+                        break
+
+                if data:
+                    data = self.get_results_data(data)
+                    if metadata:
+                        return_obj['data'] = data if data else []
+                    else:
+                        return_obj['data'] = data[int(offset): total_records] if data else []
+                    if jwt_token and next_url and result_count < self.api_client.result_limit:
+                        return_obj['metadata'] = {"result_count": result_count, "jwtToken": jwt_token,
+                                                  "next_url": next_url}
+                else:
+                    if not return_obj.get('error') and return_obj.get('success') is not False:
+                        return_obj['success'] = True
+                        return_obj['data'] = []
+            else:
+                return_obj['success'] = True
+                return_obj['data'] = []
+
+        except InvalidMetadataException as ex:
+            return_obj = self.exception_response(422, f'Invalid metadata: {str(ex)}')
+
+        except Exception as ex:
+            return_obj = self.exception_response(None, str(ex))
+        return return_obj
+
+    async def ping_connection(self):
+        """
+        Ping the endpoint
```

Comment:
Could you expand on this comment? "Ping the endpoint" doesn't really give a lot of information. Adding which endpoint is getting hit and a small blurb about it's purpose "Connects to the XXX endpoint and confirms connectivity and authentication to the prodcut".

Reply:
Updated as per the suggestion.